### PR TITLE
Automerge major dependency updates to reduce traffic

### PR DESCRIPTION
**Pull request by mcollina**
It’s important that we automerge major updates as they are the only ones that will be opened in most of our repos because we do not use lockfiles.

Let’s trust our tests to detect any potential breakage.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
cc @Fdawgs @Eomm

**Comment by Eomm** [comment]
I think this change hurt us more than the benefits it takes: mainly the plugins forward some options to external modules.
We had some issues because of that.

Since we are not testing all the external's module options, the CI could be green and merge it.
This means that _in some cases_ we should release the plugin as a major version because the options used by the dev through the fastify's plugin could be changed.

In both cases (with or without the automerge), a fastify maintainer needs to check the external dep's release to decide if it should cause a major or a minor release.

A blocked dependabot's PR is easier to manage than reading the merged PRs and going to check them.

For sure and at least, the dependabot configuration should not call these kinds of PRs as `chore`.


ref https://github.com/fastify/fastify/pull/3716

**Comment by Eomm** [comment]
Here are some examples:

- https://github.com/fastify/fastify-websocket/pull/145
- https://github.com/fastify/benchmarks/pull/228
- https://github.com/fastify/github-action-merge-dependabot/pull/74

**Comment by mcollina** [comment]
We have more than 50 modules in this or, mostly they do not have lock files and they only receive dependency updates with a semver-major. Dealing with a few  mistakes is significantly less work than manually checking all dependency updates across the full org.

Let's make it this way: I am the one that have historically done most of the releases across the org.

This changes increases the time it takes to release a module as it requires more manual steps and wait for CI runs. 

I will have to further reduce the amount of releases I can do (I already have, partially because of this). 

Are others willing to pick up the work? I would be happy to let other take the load. I get upset when others increase my workload without giving me more help.

**Comment by Eomm** [comment]
> Dealing with a few mistakes is significantly less work than manually checking all dependency updates across the full org.

Indeed, I think my approach relies less on the final users (that should open an issue) and moves the burden to the maintainer that publishes the release.

> Are others willing to pick up the work?

Yes, I'm. I think to have done it a few times already but I can do more for sure.

In any case, I would like to hear some others @fastify/plugins opinions.

**Comment by mcollina** [comment]
> Indeed, I think my approach relies less on the final users (that should open an issue) and moves the burden to the maintainer that publishes the release.

Exactly. My goal is to push as much work as possible on the users. __They are not paying for it__, doing QA and opening an issue is the minimum they can do.

**Comment by Fdawgs** [comment]
Sounds good! I agree, users should be testing updates to plugins in their applications/services using CI themselves.
If a dependency breaks a plugin, and it wasn't caught by our own CI, then hopefully users will report this back if it breaks their apps.

**Comment by Eomm** [comment]
I would propose a parameter instead of removing the `target` option.

I agree with you at some point, but I would like to define this kind of usage:

- plugin that does not forward configuration to dependencies: `target: major`
- plugin that does not forward configuration to dependencies AND have a **low** download rate: `target: major`
- plugin that forward configuration to dependencies AND have a **high** download rate `target: minor`

Adding the parameter would let us apply this setup in both ways reducing the burden.

_Notice that this workflow is used by 2 plugins only, I was planning to spread it during the `fastify@v4` upgrade_

* high download rate = 10k/week

**Comment by mcollina** [comment]
How about setting the `exclude`  settings of the automerge action? We can just skip the critical dependencies whose options are exposed to the end users.

**Comment by Fdawgs** [comment]
> 
> _Notice that this workflow is used by 2 plugins only, I was planning to spread it during the `fastify@v4` upgrade_

I have been planning on doing it, was just waiting on a general agreement on script naming in https://github.com/fastify/workflows/issues/9.

**Comment by mcollina** [comment]
@Fdawgs do you think we could add a workflow input to pass `exclude` to the automerge action?

**Comment by Fdawgs** [comment]
> @Fdawgs do you think we could add a workflow input to pass `exclude` to the automerge action?

Sure! Should have time to add it this evening.

File: .github/workflows/plugins-ci.yml
Comment (Eomm) [inline .github/workflows/plugins-ci.yml:61]: Just last request before merging: I would set `target: major` to provide more expressiveness

**Review by Fdawgs — APPROVED** [review]

**Review by Eomm — APPROVED** [review]

**Pull request merged**
